### PR TITLE
Update testing to use Cloud Build

### LINKED ISSUE #49: Update testing to use Cloud Build
We should update the testing to use Cloud Build and a new Service Account per run, as documented [here](https://docs.google.com/document/d/1eiHpZlhEDMDG8gi4LHyv2aZs1TJmeFlF8sqCE57_j4A/edit#heading=h.rz8l5qn0ygc0).

### LINKED ISSUE #51: Split lint-tests to steps for Cloud Build
Split lint-tests to each separated step for better log/results reviewing

### PR DESCRIPTION
**Pull request by paulpalamarchuk**
* Based on PR #43
 * Fix #49
 * README.md - some syntax errors fixed
 * Splitted lint-test (Fix #51)
 * Improved test fixtures

### CODE REVIEW

**Comment by paulpalamarchuk** [comment]
Kitchen tests passed:
iam-kitchen0
iam-kitchen1

File: test/setup/iam.tf [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  int_required_proj_roles = [
+    "roles/owner",
```
Comment (kopachevsky) [inline test/setup/iam.tf]: I think roles/owner overlaps with lots of roles below?

Comment (ingwarr) [reply test/setup/iam.tf]: I don't think so

Comment (paulpalamarchuk) [reply test/setup/iam.tf]: No it's not

**Comment by paulpalamarchuk** [comment]
@aaron-lane rebased

File: test/fixtures/full/base/outputs.tf [hunk 2]
```diff
@@ -72,3 +72,14 @@ output "region" {
   value       = local.location
   description = "Subnetworks region created for bindings."
 }
+
+output "fixture_project_id" {
```
Comment (aaron-lane) [inline test/fixtures/full/base/outputs.tf]: An output which forwards an input is an anti-pattern; the input should be used directly in the calling module.

File: test/fixtures/full/iam.tf [hunk 3]
```diff
@@ -45,57 +49,72 @@ module "iam_binding_subnet" {
...
-  source               = "../../../modules/pubsub_subscriptions_iam"
-  mode                 = var.mode
-  pubsub_subscriptions = module.base.subscriptions
-  project              = var.fixture_project_id
-
-  bindings = local.basic_bindings
+  source                   = "../../../modules/pubsub_subscriptions_iam"
+  mode                     = var.mode
+  pubsub_subscriptions     = module.base.subscriptions
+  project                  = module.base.fixture_project_id
```
Comment (aaron-lane) [inline test/fixtures/full/iam.tf]: This should use `var.project_id`.

File: test/fixtures/full/iam.tf [hunk 4]
```diff
@@ -45,57 +49,72 @@ module "iam_binding_subnet" {
...
-  source        = "../../../modules/pubsub_topics_iam"
-  mode          = var.mode
-  pubsub_topics = module.base.topics
-  project       = var.fixture_project_id
-
-  bindings = local.basic_bindings
+  source            = "../../../modules/pubsub_topics_iam"
+  mode              = var.mode
+  pubsub_topics     = module.base.topics
+  project           = module.base.fixture_project_id
```
Comment (aaron-lane) [inline test/fixtures/full/iam.tf]: This should use `var.project_id`.

File: test/setup/iam.tf (hunk 1 above)
Comment (aaron-lane) [reply test/setup/iam.tf]: Actually, most of these roles are completely covered by the owner role, but the distinction of the roles in the README are also ambiguous so let's leave it for now.

File: test/setup/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+output "project_id" {
+  value = module.iam-project.project_id
+}
+
+output "sa_key" {
+  value     = google_service_account_key.int_test.private_key
+  sensitive = true
+}
+
+output "assigned_project_roles" {
```
Comment (aaron-lane) [inline test/setup/outputs.tf]: This output is not used; it can be dropped.

File: test/fixtures/full/base/main.tf [hunk 6]
```diff
@@ -42,10 +42,9 @@ resource "google_project" "test" {
   count = local.n
```
Comment (aaron-lane) [inline test/fixtures/full/base/main.tf]: I hit a race condition where `google_storage_bucket.test` failed to apply due to this project not being ready. I think we should attach all resources to the project created in `test/setup` and only use this for testing project IAM bindings. Both `google_project_service` resources should also be removed as they are already enabled on the base project.

Comment (aaron-lane) [reply test/fixtures/full/base/main.tf]: This should resolve the error in CI.

Comment (paulpalamarchuk) [reply test/fixtures/full/base/main.tf]: Moved SA creation to the "preparation" step, now CI tests runs well

**Pull request merged**